temporalcli/commands.workflow_exec.go

```diff
@@ -0,0 +1,429 @@
+		RunTime string `cli:",cardOmitEmpty"`
+		Status  string
+		Result  json.RawMessage `cli:",cardOmitEmpty"`
+		Failure string          `cli:",cardOmitEmpty"`
+	}{}
+	if duration > 0 {
+		result.RunTime = duration.Truncate(10 * time.Millisecond).String()
+	}
+	switch closeEvent.EventType {
+	case enums.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
+		result.Status = color.GreenString("COMPLETED")
+		var err error
+		result.Result, err = cctx.MarshalFriendlyJSONPayloads(
+			closeEvent.GetWorkflowExecutionCompletedEventAttributes().GetResult())
+		if err != nil {
+			return fmt.Errorf("failed marshaling result: %w", err)
+		}
+	case enums.EVENT_TYPE_WORKFLOW_EXECUTION_FAILED:
+		result.Status = color.RedString("FAILED")
+		result.Failure = cctx.MarshalFriendlyFailureBodyText(
+			closeEvent.GetWorkflowExecutionFailedEventAttributes().Failure, "    ")
+	case enums.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT:
+		result.Status = color.RedString("TIMEOUT")
+	case enums.EVENT_TYPE_WORKFLOW_EXECUTION_CANCELED:
+		result.Status = color.RedString("CANCELED")
+	}
+	return cctx.Printer.PrintStructured(result, printer.StructuredOptions{})
+}
+
+func (c *TemporalWorkflowCommand) startWorkflow(
+	cctx *CommandContext,
+	cl client.Client,
+	workflowOpts *WorkflowStartOptions,
+	inputOpts *PayloadInputOptions,
+	printRunningExecutionEvenWithJSON bool,
+) (client.WorkflowRun, error) {
+	startOpts, err := workflowOpts.buildStartOptions()
+	if err != nil {
+		return nil, err
+	}
+	input, err := inputOpts.buildRawInput()
+	if err != nil {
+		return nil, err
+	}
+	run, err := cl.ExecuteWorkflow(cctx, startOpts, workflowOpts.Type, input...)
+	if err != nil {
+		return nil, fmt.Errorf("failed starting workflow: %w", err)
+	}
+
+	// Print running execution
+	if !cctx.JSONOutput || printRunningExecutionEvenWithJSON {
+		cctx.Printer.Println(color.MagentaString("Running execution:"))
+		err := cctx.Printer.PrintStructured(struct {
+			WorkflowId string `json:"workflowId"`
+			RunId      string `json:"runId"`
+			Type       string `json:"type"`
+			Namespace  string `json:"namespace"`
+			TaskQueue  string `json:"taskQueue"`
+		}{
+			WorkflowId: run.GetID(),
+			RunId:      run.GetRunID(),
+			Type:       workflowOpts.Type,
+			Namespace:  c.Namespace,
+			TaskQueue:  workflowOpts.TaskQueue,
+		}, printer.StructuredOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed printing: %w", err)
+		}
+	}
+	return run, nil
+}
+
+func (w *WorkflowStartOptions) buildStartOptions() (client.StartWorkflowOptions, error) {
+	o := client.StartWorkflowOptions{
+		ID:                                       w.WorkflowId,
+		TaskQueue:                                w.TaskQueue,
+		WorkflowRunTimeout:                       w.RunTimeout,
+		WorkflowExecutionTimeout:                 w.ExecutionTimeout,
+		WorkflowTaskTimeout:                      w.TaskTimeout,
+		CronSchedule:                             w.Cron,
+		WorkflowExecutionErrorWhenAlreadyStarted: !w.AllowExisting,
```

Comment:
Just to confirm this is the default behaviour here is the existing behaviour of the CLI?

Reply:
Oh sorry I missed that note. I may have missed that discussion. I would keep the behaviour the same as the old CLI and add an option to not allow existing instead:
* It matches our most popular SDKs behaviour
* Makes migrating to the new CLI easier

Reply:
:+1: Will bring it up internally for discussion, but I am kinda leaning your direction too. If we never had a previous CLI, I would hard-lean towards start failing if...well...start fails.

Reply:
Done